app/controllers/volunteers_controller.rb

```diff
@@ -26,7 +27,15 @@ def create
 
   def update
     if @volunteer.update(volunteer_params)
-      redirect_to @volunteer, notice: t('volunteer_updated')
+      if @state == 'interested/registered' && @volunteer.state == 'active'
+        new_user = User.new(email: @volunteer.email,
+          password: Devise.friendly_token, role: 'volunteer')
+        new_user.save!
+        new_user.invite!
```

Comment:
In the create action of the users controller, a user with password is created before user.save and user.invite